android/app/src/main/java/org/fossasia/openevent/viewmodels/LocationsFragmentViewModel.java

```diff
@@ -37,4 +39,9 @@ public void setSearchText(String searchText) {
         this.searchText = searchText;
     }
 
+    @Override
+    protected void onCleared() {
+        filteredLocations.disposeCompositeDisposable();
```

Comment:
This should be handled by the LiveData itself. That's the point of it being LiveData, it is LifeCycle aware